pyladie_oop/pylady.py

```diff
@@ -0,0 +1,103 @@
+class PyLady:
+
+    def __init__(self, name):
+        self.name = name
+        self.knowledge = 0  # max je 5
+        self.max_knowledge = 5
+        self.practice = 0  # max je 5
+        self.max_practice = 5
+        self.focus = 3  # minimum 0
+        self.max_focus = 3
+        self.energy = 5  # minimum 0
+        self.max_energy = 5
+        self.coach_capacity = 3  # min je 0
+        self.max_coach_capacity = 3
+
+    def __str__(self):
+        return f"PyLady {self.name}. Její practice je {self.graficky_practice()} a knowledge {self.graficky_knowledge()}"
+
+    def graficky_knowledge(self):
+        celkem = int(self.max_knowledge)
+        pocet = int(self.knowledge)
+        return "[{0}{1}]".format("#"*pocet, "_" * (celkem-pocet))
+
+    def graficky_practice(self):
+        celkem = int(self.max_practice)
+        pocet = int(self.practice)
+        return "[{0}{1}]".format("#"*pocet, "_" * (celkem-pocet))
+
+    def study(self, hour):
```

Comment:
Tu s energy a focus pracuješ tak, že hneď na začiatku odpočítaš energy/focus odpovedajúcu hodinám a ak ti klesne do mínusu tak ju zmeníš na 0. To by ale reálne nefungovalo, nie? Nemôžeš študovať 8 hodín keď máš energiu len na 5, tým pádom sa ti ani knowledge a practice nemôžu zvýšiť o 8/20, resp 8/10 u practice. Ako by si to upravila aby to čerpalo energiu a focus postupne a a teda ja postupne pripisovalo knowledge a practice až do chvíle kedy budeš na 0 v aspoň jednom z energy a focus?